src/editors/EditorHeader.jsx

```diff
@@ -1,27 +1,82 @@
 import {
-  ActionRow, IconButton, Icon, ModalDialog,
+  ActionRow, IconButton, Icon, ModalDialog, Form, Row,
 } from '@edx/paragon';
 import PropTypes from 'prop-types';
-import React, { useContext } from 'react';
-import { Close } from '@edx/paragon/icons';
+import React, { useContext, useState, useEffect } from 'react';
+import { Edit, Close } from '@edx/paragon/icons';
 import EditorPageContext from './EditorPageContext';
-import { ActionStates, mapBlockTypeToName } from './data/constants';
+import { ActionStates } from './data/constants';
 
-const EditorHeader = ({ title }) => {
-  const { unitUrl, unitUrlLoading, studioEndpointUrl } = useContext(EditorPageContext);
+const EditorHeader = () => {
+  const {
+    titleRef, editorRef, setBlockTitle,
+    blockValue, blockLoading,
+    unitUrl, unitUrlLoading, studioEndpointUrl 
+  } = useContext(EditorPageContext);
+
+  const [editing, setEditing] = useState(false);
+
+  const [title, setTitle] = useState("");
+  useEffect(() => {
+    if (blockLoading === ActionStates.FINISHED) {
+      setTitle(blockValue ? blockValue.data.display_name : '');
+      setEditing(true);
+    }
+  }, [blockLoading]);
+
+  const handleChange = (e) => {
+    setTitle(e.target.value);
+  };
+
+  const handleKeyDown = (e) => {
+    if (e.key === 'Enter') setEditing(false);
+    if (e.key === 'Tab' && editorRef) {
+      e.preventDefault();
+      editorRef.current.focus();
+    }
+  };
```

Comment:
When user is editing title, they will exit the editing process if they press 'Enter'. If they press tab, the cursor should move on to the editor.